### DISCUSSION TITLE
How to create a bitcoin P2SH address？

### DISCUSSION THREAD

**Discussion by YaakovHuang** [trustwallet/wallet-core#1358] [Q&A]
How to create a bitcoin P2SH address, I did not find a relevant method on the document, I am using android, thx.

**Comment by optout21** [comment]
There is this TW C method, it is available from Kotlin as well:

https://github.com/trustwallet/wallet-core/blob/master/src/interface/TWBitcoinAddress.cpp#L31

**Reply by YaakovHuang** [reply]
I still don’t quite understand, how should I create this P2SH address if I use Kotlin?

**Reply by hewigovens** [reply]
please checkout kotlin tests: https://github.com/trustwallet/wallet-core/blob/master/android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/bitcoin/TestBitcoinScript.kt, to build a P2SH address, you need to build the script by yourself, and there is a java `BitcoinAddress` class available for kotlin
